crates/ruma-common/src/identifiers/key_id.rs

```diff
@@ -43,7 +43,8 @@ impl<A: KeyAlgorithm, K: KeyName + ?Sized> KeyId<A, K> {
     where
         &'a K: TryFrom<&'a str>,
     {
-        <&'a K>::try_from(&self.as_str()[..self.colon_idx()]).unwrap_or_else(|_| unreachable!())
+        <&'a K>::try_from(&self.as_str()[(self.colon_idx() + 1)..])
+            .unwrap_or_else(|_| unreachable!())
```

Comment:
As an aside: it wasn't at all clear to me what the `KeyId::key_name` was actually supposed to return. What exactly is the "key name" of a `DeviceKeyId`? https://docs.rs/ruma/latest/ruma/struct.KeyId.html#method.key_name is unhelpful.

I don't think "key name" is a term that is used in the Matrix Spec. https://spec.matrix.org/v1.12/appendices/#signing-details uses the words "key identifier" for this concept:

> The *signing key identifier* is the concatenation of the *signing algorithm* and a *key identifier*.

... though I don't think that would be very clear either.

Come to that, it took me quite a while to figure out what a `DeviceKeyId` was supposed to represent.

I feel like examples would be worth a thousand words here. Maybe one of those tests below could actually be a doctest?

In short, could we improve the documentation here while we're in the ballpark?

Reply:
I started on a PR to rename `key_name` to `key_identifier` then I wondered whether `KeyId` should be `SigningKeyIdentifier`, started on that, and found that there is already a `SigningKeyId` so this appears to be above my pay grade.

Reply:
I am working on some docs.

Reply:
This type is not only used for signing key IDs. It is also used for one-time and fallback key IDs, and the device Curve25519 identity key ID, which is why they are separate types.

Reply:
@richvdh I could do with some help on the wording here: https://github.com/andybalaam/ruma/pull/1